File: code/read/jdbc/src/main/java/be/rubus/microstream/performance/jdbc/tests/PurchaseOfForeigners.java
package be.rubus.microstream.performance.jdbc.tests;

import be.rubus.microstream.performance.QueryInformation;
import be.rubus.microstream.performance.StopWatch;
import be.rubus.microstream.performance.jdbc.Service;
import be.rubus.microstream.performance.jdbc.model.Purchase;
import be.rubus.microstream.performance.utils.DurationUtil;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.List;

public class PurchaseOfForeigners {

    public static void main(String[] args) throws ClassNotFoundException {
        Logger logger = LoggerFactory.getLogger(PurchaseOfForeigners.class);

        logger.info("Performance run JDBC");
        StopWatch stopWatch = StopWatch.start();

        Class.forName("org.postgresql.Driver");

        try (Connection connection = DriverManager
                .getConnection("jdbc:postgresql://localhost:5432/postgres", "postgres", "mysecretpassword")) {

            DurationUtil.printDuration(logger, "Database Access", stopWatch.stop());

            Service service = new Service(connection);

            logger.info("Execute Query - PurchaseOfForeigners");
            QueryInformation<List<Purchase>> listQueryInformation = service.purchaseOfForeigners();

            DurationUtil.printDuration(logger, "Query Execution", listQueryInformation.getElapsedNanoSeconds());

        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}

File: code/read/exposed/src/main/kotlin/be/rubus/microstream/performance/exposed/tests/PurchaseOfForeigners.kt
package be.rubus.microstream.performance.exposed.tests

import be.rubus.microstream.performance.StopWatch
import be.rubus.microstream.performance.exposed.DBSettings
import be.rubus.microstream.performance.exposed.Service
import be.rubus.microstream.performance.exposed.readonlyTransaction
import be.rubus.microstream.performance.utils.DurationUtil
import org.jetbrains.exposed.sql.StdOutSqlLogger
import org.jetbrains.exposed.sql.addLogger
import org.slf4j.LoggerFactory

fun main() {
    val logger = LoggerFactory.getLogger("PurchaseOfForeigners")
    logger.info("Performance run Exposed")

    val stopWatch: StopWatch = StopWatch.start()
    DBSettings.db
    DurationUtil.printDuration(logger, "Database Access", stopWatch.stop())

    logger.info("Execute Query - Purchase Of Foreigners")
    val listQueryInformation = readonlyTransaction(DBSettings.db) {
        //addLogger(StdOutSqlLogger)
        Service.purchaseOfForeigners()
    }

    DurationUtil.printDuration(logger, "Query Execution", listQueryInformation.elapsedNanoSeconds)


}

